clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -230,6 +227,14 @@ private ClientRequest makeClientRequest(
             unsent.handler
         );
     }
+    
+    public Optional<Exception> metadataError() {
+        return metadataError;
+    }
+    
+    public void clearMetadataError() {
+        metadataError = Optional.empty();
+    }
```

Comment:
Then this can be a single operation:

```suggestion
    public Optional<Exception> getAndClearMetadataError() {
        Optional<Exception> e = metadataError;
        metadataError = Optional.empty();
        return e;
    }
```